### ISSUE TITLE
Sync raw sensor data from Node to Hub

### ISSUE THREAD

**Issue by JamieLaing** [netglass-io/NobleOne-Releases#67]
## Overview

Implement sync of raw CAN and IMU sensor data from Node to Hub. This enables driver performance analysis and event detection algorithms on Hub.

**Parent:** #43 (Driver Performance & Gamification EPIC)

## Current State

**✅ Already implemented:**
- SQLite storage of raw sensor data on Node (255K+ records tested)
- `RawSensorData` table with CAN and IMU data
- `DriverSessions` linked to sensor data
- `SyncedToHubAt` timestamp field in schema
- Core sync infrastructure (`SyncQueueService`, `HubSyncBackgroundService`)

**❌ Missing:**
- Event type for raw sensor data batches
- Hub API endpoint to receive sensor data
- Export/packaging logic on Node
- Retention/cleanup after successful sync

## Architecture

Per `/Node/Docs/DATA_SYNC_ARCHITECTURE.md`:
- **Transport:** REST API (bulk historical data, not SignalR)
- **Pattern:** Queue with Confirmed Delivery (critical training data)
- **Batch size:** 1000 records per request (configurable)

### Phase 1: Node Export & Queue (Node repo)

**1. Create export service** (`Services/SensorDataExportService.cs`):
- Query completed sessions with labeled data
- Package sensor data with metadata (truck, driver, timestamps)
- Batch records (1000 per request)
- Queue via `SyncQueueService.QueueEventAsync("RawSensorDataBatch", batch)`

**2. Update HubSyncBackgroundService**:
- Add handler for `RawSensorDataBatch` event type
- POST to Hub REST API: `/api/bulk/sensor-data`
- Mark records as synced on success

**3. Add retention/cleanup logic**:
- Update `RawSensorData.SyncedToHubAt` timestamp on successful sync
- Periodic cleanup job: delete synced records older than 30 days
- Configurable retention policy via settings

### Phase 2: Hub API Endpoint (Hub repo)

**1. Create bulk data controller** (`Controllers/BulkDataController.cs`):
- Endpoint: `POST /api/bulk/sensor-data`
- Accept batch with validation
- Store in Hub database for ML training
- Return success/failure status

**2. Hub database schema**:
- Store raw sensor data with session metadata
- Index by truck, driver, timestamp for query performance
- Link to Hub's driver and truck records

**3. Deduplication logic**:
- Check for existing records by Node ID + timestamp
- Skip duplicates to handle retry scenarios

### Phase 3: Export UI & Monitoring (Node repo)

**1. Export management UI** (`/settings/export`):
- List completed sessions available for export
- Manual "Export to Hub" button (testing/fallback)
- Export status indicator (pending, syncing, synced)
- Error logs for failed exports

**2. Monitoring dashboard**:
- Total records synced
- Pending export queue size
- Last successful sync timestamp
- Sync errors and retry count

## Acceptance Criteria

- [ ] Node can package and queue completed sessions for Hub upload
- [ ] Hub API endpoint receives and stores sensor data batches
- [ ] Successful sync updates `RawSensorData.SyncedToHubAt` timestamp
- [ ] Old synced data automatically cleaned up after 30 days
- [ ] Export status visible in Node UI
- [ ] Manual export trigger available for testing
- [ ] Deduplication prevents duplicate records on retry
- [ ] Works offline (queues until Hub connection restored)

## Testing Plan

1. **Basic export test:**
   - Complete a driver session on Node
   - Verify data queued for export
   - Confirm successful upload to Hub
   - Verify `SyncedToHubAt` timestamp updated

2. **Offline recovery test:**
   - Disable Hub connection
   - Complete multiple sessions
   - Restore connection
   - Verify all sessions sync successfully

3. **Retention test:**
   - Sync old test data
   - Fast-forward system time or manually trigger cleanup
   - Verify old synced records deleted

4. **Large batch test:**
   - Record session with 10K+ sensor records
   - Verify batching (multiple 1000-record uploads)
   - Confirm all records received by Hub

## Dependencies

- SQLite raw sensor storage (#62) - **COMPLETE**
- Hub API authentication/authorization
- Hub database schema for sensor data storage

## Out of Scope

- Real-time sensor streaming (use SignalR per architecture, separate feature)
- ML model training algorithms (separate epic)
- Compression/optimization (future enhancement)
- Session labeling UI (separate issue)

## Related Documentation

- `/Node/Docs/DATA_SYNC_ARCHITECTURE.md`
- `/Node/handoff.md` (Issue #62 TODO section)

## Priority

**Medium** - Foundation for driver performance ML models. Not blocking current operations but required for future analytics.

**Comment by JamieLaing** [comment]
**Update:** `LabeledSessions` table added - include in sync scope.

New table for driver-created labeled recording segments:
- Driver enters label like "putting away one pallet"
- Hits Start → sensor data gets tagged with `LabeledSessionId`
- Hits Stop → segment complete

Schema:
```sql
LabeledSessions:
  Id, Label, Description, StartedAt, EndedAt,
  DriverSessionId, TruckId, SyncedToHubAt
```

RawSensorData now has `LabeledSessionId` FK (nullable).

**Sync requirement:** Include `LabeledSessions` table in sync to Hub alongside `RawSensorData`. The labeled sessions contain the training data metadata.
